### PR TITLE
Fix parsing of query with nested object

### PR DESCRIPTION
**Pull request by colincasey**
This modifies the parsing routine of data api queries to correctly parse result sets that include nested `Record` data.

Requires the following SDK change:
* https://github.com/forcedotcom/sf-fx-sdk-java/pull/106

[W-12157391](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001FgZMwYAN/view)

### CODE REVIEW

**Review by edmorley — COMMENTED** [review]
Cancelling review since CI is failing

File: sf-fx-runtime-java-sdk-impl-v1.1/src/test/java/com/salesforce/functions/jvm/runtime/sdk/DataApiImplTest.java [hunk 1]
```diff
@@ -346,4 +346,18 @@ public void testGetAccessToken() {
...
+    RecordQueryResult result = dataApi.query("SELECT Name, Owner.Name from Account LIMIT 1");
+    assertThat(result.isDone(), is(true));
+    assertThat(result.getTotalSize(), is(1L));
+
+    Record record = result.getRecords().get(0);
+    assertThat(record.getRecordField("Owner").get(), isA(Record.class));
+
+    Record nestedRecord = record.getRecordField("Owner").get();
+    assertThat(nestedRecord.getType(), is("User"));
+    assertThat(nestedRecord.getStringField("Name").get(), is("Guy Smiley"));
```
Comment (Malax) [inline sf-fx-runtime-java-sdk-impl-v1.1/src/test/java/com/salesforce/functions/jvm/runtime/sdk/DataApiImplTest.java:361]: "Did somebody say air?"

**Comment by Malax** [comment]
CI is failing due to sf-fx-sdk-java 1.1.1 not being released yet. After https://github.com/forcedotcom/sf-fx-sdk-java/pull/106 is merged an released, CI for this PR should start working again.

**Comment by edmorley** [comment]
@Malax Is there a way the WIP over there can be tested here? It seems suboptimal to not be able to see passing CI before releasing things? Would merging that repository into this one (whilst still keeping the packages separate) help?

**Comment by Malax** [comment]
@edmorley, moving the contents here is probably more complicated than installing the dependency "locally" in CI (`./mvnw install`). We could add this to the CI scripts for each unreleased version and the git ref that needs to be installed.

**Review by Malax — CHANGES_REQUESTED** [review]
Do we know if functions that depend on `sf-fx-sdk-java` in version `1.1.0` still work with the updated SDK 1.1 impl? If not, we probably need a separate SDK impl for `1.1.1`. On that note, I think the version mapping from SDK version to implementation JAR is missing for `1.1.1`. See `SalesforceFunctionsProjectFunctionsScanner`.

File: sf-fx-runtime-java-sdk-impl-v1.1/src/main/java/com/salesforce/functions/jvm/runtime/sdk/AbstractRecordAccessorImpl.java [hunk 2]
```diff
@@ -139,12 +128,34 @@ public Optional<ByteBuffer> getBinaryField(String name) {
   }
 
   @Nonnull
-  private <T, E extends Throwable> Optional<T> getJsonFieldValue(
-      String fieldName, Function<JsonPrimitive, T> f, E binaryDataException) throws E {
+  @Override
+  public Optional<Record> getRecordField(String name) {
```
[the remark this answers is not in the document]
Comment (colincasey) [reply sf-fx-runtime-java-sdk-impl-v1.1/src/main/java/com/salesforce/functions/jvm/runtime/sdk/AbstractRecordAccessorImpl.java:132]: naming does not have to be consistent with `getJsonFieldValue` which is **private**.  it has to be consistent with the **public SDK** methods from `RecordAccessor` such as:
* `getStringField`
* `getBinaryField`
* and so on...

hence, `getRecordField` 😄

[the remark this answers is not in the document]
Comment (Malax) [reply sf-fx-runtime-java-sdk-impl-v1.1/src/main/java/com/salesforce/functions/jvm/runtime/sdk/AbstractRecordAccessorImpl.java:132]: That makes total sense, I didn't see that it's not another internal helper. Sorry for the noise! :)

**Review by Malax — CHANGES_REQUESTED** [review]
Implementation-weise this is good to go IMO. Do we want to add an integration test for a `1.1.1` SDK function as a smoke test? Seems valuable to me.

Next steps required before we can ship/approve this PR:
- Merge and release https://github.com/forcedotcom/sf-fx-sdk-java/pull/106 (happy to assist if this is your first time releasing)
- Update `sf-fx-sdk-java`'s `1.1.1-SNAPSHOT` version to `1.1.1` in this PR
- Verify that CI passes

**Review by Malax — APPROVED** [review]

**Pull request merged**